stackslib/src/chainstate/nakamoto/staging_blocks.rs

```diff
@@ -664,7 +673,7 @@ impl StacksChainState {
             Ok(x) => x,
             Err(e) => {
                 debug!("Failed to get Nakamoto staging blocks DB version: {:?}", &e);
-                return Ok(1);
+                return Ok(NAKAMOTO_STAGING_DB_SCHEMA_LATEST);
```

Comment:
This needs to be reverted, because the table `db_version` didn't exist in schema 1 (which is why this error path was written the way it was).

Reply:
Interesting - yeah that makes sense. In that case, we'd need to handle the difference between an error caused by the table simply not existing, vs some other error (like a connection error). What we don't want to do (which is what was brought up in the audit) is to execute `DROP TABLE nakamoto_staging_blocks` because there was some intermittent error in the above `query_row`.

Reply:
I've added a check to see if the `db_version` table exists - if not, it returns version 1. Otherwise, "regular" DB errors will cause this function to return an `Err` instead of defaulting to 1.